### PR TITLE
Remove redundant AI score increases

### PR DESCRIPTION
**Pull request by AlexOn1ine**
#### All of those are handled by damage calcs (added tests for those)
- EFFECT_FACADE
- EFFECT_WAKE_UP_SLAP
- EFFECT_SMELLINGSALT
- EFFECT_GRAV_APPLE
- EFFECT_FLAIL
- EFFECT_DREAM_EATER
- EFFECT_SPIT_UP

#### These are debatable but I don't think they should be increased based on some value that isn't taking damage into account
- EFFECT_ABSORB
- EFFECT_REVENGE
- EFFECT_CONFUSE_HIT
- EFFECT_ROLLOUT

Also debatable but I removed `// attacker ability checks` because I don't think it's necessary to check since AI will go for a killing move if it sees a kill anyways.

### CODE REVIEW

**Comment by DizzyEggg** [comment]
This branch is out-of-date with the base branch
Merge the latest changes from upcoming into this branch.

File: src/battle_ai_main.c [hunk 1]
```diff
@@ -3295,63 +3291,16 @@ static s32 AI_CheckViability(u32 battlerAtk, u32 battlerDef, u32 move, s32 score
...
-                score = 90; // Force switch if all your attacking moves are physical and you have Natural Cure.
-            break;
-        default:
-            if (IS_MOVE_PHYSICAL(move) && gBattleMoves[move].effect != EFFECT_FACADE)
-                ADJUST_SCORE(-2);
-            break;
-        }
+        if ((gBattleMons[battlerAtk].status1 & STATUS1_BURN && HasOnlyMovesWithSplit(battlerAtk, SPLIT_PHYSICAL, TRUE))
+        || (gBattleMons[battlerAtk].status1 & STATUS1_FROSTBITE && HasOnlyMovesWithSplit(battlerAtk, SPLIT_SPECIAL, TRUE)))
+            score = 90; // Force switch if all your attacking moves are physical and you have Natural Cure.
```
Comment (DizzyEggg) [inline src/battle_ai_main.c:3301]: Oh I must have missed that, but it should use the SET_SCORE macro

File: src/battle_ai_main.c [hunk 2]
```diff
@@ -3362,12 +3311,6 @@ static s32 AI_CheckViability(u32 battlerAtk, u32 battlerDef, u32 move, s32 score
         if (AI_RandLessThan(128))
             IncreaseSleepScore(battlerAtk, battlerDef, move, &score);
         break;
-    case EFFECT_ABSORB:
```
Comment (DizzyEggg) [inline src/battle_ai_main.c:3365]: Why did you remove the score for Big Root? This isn't part of the damage calc.

File: test/battle/ai_check_viability.c [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+    GIVEN {
+        ASSUME(gBattleMoves[MOVE_GRAV_APPLE].effect == EFFECT_GRAV_APPLE);
+        ASSUME(gBattleMoves[MOVE_TROP_KICK].effect == EFFECT_ATTACK_DOWN_HIT);
+        AI_FLAGS(AI_FLAG_CHECK_BAD_MOVE | AI_FLAG_CHECK_VIABILITY | AI_FLAG_TRY_TO_FAINT);
+        PLAYER(SPECIES_WOBBUFFET) { HP(81); Speed(20); }
+        OPPONENT(SPECIES_WOBBUFFET) { Speed(10); Moves(MOVE_TROP_KICK, MOVE_GRAV_APPLE); }
+    } WHEN {
+        TURN { MOVE(player, movePlayer); EXPECT_MOVE(opponent, MOVE_TROP_KICK); }
+        TURN { MOVE(player, MOVE_CELEBRATE);EXPECT_MOVE(opponent, expectedMove); }
```
Comment (DizzyEggg) [inline test/battle/ai_check_viability.c:88]: TURN { MOVE(player, MOVE_CELEBRATE); EXPECT_MOVE(opponent, expectedMove); }

File: test/battle/ai_check_viability.c [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+        AI_FLAGS(AI_FLAG_CHECK_BAD_MOVE | AI_FLAG_CHECK_VIABILITY | AI_FLAG_TRY_TO_FAINT);
+        PLAYER(SPECIES_WOBBUFFET) { HP(43); }
+        OPPONENT(SPECIES_WOBBUFFET) { Moves(MOVE_STOCKPILE, MOVE_SPIT_UP, MOVE_TACKLE); }
+    } WHEN {
+        TURN { EXPECT_MOVE(opponent, MOVE_STOCKPILE); }
+        TURN { EXPECT_MOVE(opponent, MOVE_SPIT_UP); }
+    } SCENE {
+        // MESSAGE("Wobbuffet fainted!");
+    }
+}
```
Comment (DizzyEggg) [inline test/battle/ai_check_viability.c:145]: newline missing

**Review by DizzyEggg — CHANGES_REQUESTED** [review]
Nice work with the AI tests :D

File: src/battle_ai_main.c (hunk 2 above)
Comment (DizzyEggg) [reply src/battle_ai_main.c:3365]: That should work

File: test/battle/ai_check_viability.c (hunk 4 above)
Comment (AlexOn1ine) [reply test/battle/ai_check_viability.c]: added

**Comment by AlexOn1ine** [comment]
> This branch is out-of-date with the base branch Merge the latest changes from upcoming into this branch.

Updated branch and applied fixes.

**Review by DizzyEggg — APPROVED** [review]

**Comment by DizzyEggg** [comment]
Oh, CI failed 👀

**Review by DizzyEggg — APPROVED** [review]
